### PR TITLE
feat: Make use of new "async" RPC

### PR DESCRIPTION
**Pull request by L3tum**
| Q             | A
| ------------- | ---
| Bugfix?       | ❌
| Breaks BC?    | ❌ 
| New feature?  | ✔️TODO: 
| Issues        | https://github.com/roadrunner-php/goridge/pull/22
| Docs PR       | spiral/docs#... 

This is a follow-up PR for https://github.com/roadrunner-php/goridge/pull/22 implementing the new "async" RPC calls in the Cache class.
~It may be an option to instead modify the Factory to return an AsyncCache class instead if it detects an async RPC interface, or something like that. Please lmk if you'd like me to change that. Once the implementation is good to go I'll add tests for it as well.~

EDIT: FYI I switched to the factory (similar to the other PR in Metrics) because I think this is easier to use. I've also added the corresponding tests.

For further info please read the linked PR in the goridge repo.

Same PR for the metrics repo: https://github.com/roadrunner-php/metrics/pull/13

## Summary by CodeRabbit

- **New Features**
  - Introduced asynchronous cache operations, including methods for setting, deleting, and committing cache data asynchronously.

- **Bug Fixes**
  - Improved version compatibility for dependencies, ensuring smoother integration with `spiral/roadrunner` and `spiral/goridge`.

- **Tests**
  - Added comprehensive test cases for the new asynchronous cache functionality, covering various scenarios and error handling.

- **Chores**
  - Updated dependency version constraints in `composer.json` for better alignment and compatibility.

### CODE REVIEW

**Comment by rustatian** [comment]
@L3tum @msmakouz Friendly ping, what is a showstopper here? 😃

**Comment by L3tum** [comment]
Hey @rustatian, sorry I've been kinda busy as I'm currently hunting for a new apartment and my workplace is in a complete overhaul. There shouldn't be anything left to do here (and it's already in use) but lmk if there's any issues and I'll try to fix them

**Comment by rustatian** [comment]
Hey @L3tum 👋 Glad to hear you 👍 
It would be great if you would be able to finish this PR in terms of CI and updating deps (since we've released the worker package with your changes).

**Comment by L3tum** [comment]
Hey @rustatian / @msmakouz , PR is fixed now and I've upped the required version for goridge to 4.2 (which is the first version that has the async changes). It could be made to work with prior versions by moving the usings I think, so lmk if it's okay with you with how it is right now or if it needs to be changed.

**Comment by msmakouz** [comment]
> Hey @rustatian / @msmakouz , PR is fixed now and I've upped the required version for goridge to 4.2 (which is the first version that has the async changes). It could be made to work with prior versions by moving the usings I think, so lmk if it's okay with you with how it is right now or if it needs to be changed.

Hey @L3tum, raising the minimum version of spiral/goridge to 4.2 is normal. This is a minor version increase, and there is no problem with that.

File: composer.json [hunk 1]
```diff
@@ -48,8 +48,9 @@
         "ext-json": "*",
         "psr/simple-cache": "2 - 3",
         "roadrunner-php/roadrunner-api-dto": "^1.0",
+        "spiral/roadrunner": "^2023.1",
```
Comment (msmakouz) [inline composer.json]: @L3tum Adding `spiral/roadrunner: ^2023.1` on line 51 looks incorrect. This package is on line 53.

**Review by msmakouz — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (L3tum) [reply composer.json]: Seems to have come in with the merge from master. Deleted it :)

**Review by msmakouz — APPROVED** [review]

**Pull request merged**